integration/query_compat_test.go

```diff
@@ -90,7 +91,13 @@ func testQueryCompat(t *testing.T, testCases map[string]queryCompatTestCase) {
 					var explainRes bson.D
 					require.NoError(t, targetCollection.Database().RunCommand(ctx, explainQuery).Decode(&explainRes))
 
-					assert.Equal(t, tc.resultPushdown, explainRes.Map()["pushdown"])
+					// temporarily make possible to skip tigris pushdown checks
+					if tc.skipTigrisPushdown && setup.IsTigris(t) {
+						require.True(t, tc.resultPushdown, "Cannot use skipTigrisPushdown when resultPushdown is false'")
+						assert.Equal(t, false, explainRes.Map()["pushdown"], "Tigris pushdown check was skipped, but it was actually true")
+					} else {
+						assert.Equal(t, tc.resultPushdown, explainRes.Map()["pushdown"])
+					}
```

Comment:
That part is not really clear to me. The comment above says we want to skip Tigris pushdown checks but we still have some asserts.

Could it be simplified to something like this?

```suggestion
					if !(tc.skipTigrisPushdown && setup.IsTigris(t)) {
						assert.Equal(t, tc.resultPushdown, explainRes.Map()["pushdown"])
					}
```

Reply:
This simplification removes some assumptions:
1. We shouldn't skip pushdown checks for tigris if we don't expect any pushdowns (if `tc.resultPushdown` is not set)
2. We shouldn't skip pushdown checks for tigris if we actually do a pushdown for tigris in a testcase

The `tc.skipTigrisPushdown` was introduced to make possible introducing PostgreSQL pushdowns without Tigris ones, as they seem more complicated.

Reply:
Although I can see some inconsistency with the naming, right. When we use `skipForTigris`, we just want to skip the whole test regardless of whether it pass or fails, and here we validate if our expectations are correct. Although, I couldn't find any better name... `dontDoTigrisPushdown`, `tigrisNotPushdowned`... Any ideas? :smile: 

We could split `resultPushdowns` to `pushdownPg` and `pushdownTrigris`, but imo in the future it will create too many lines, as we aim to do pushdowns for both handlers, so personally I would stay with current approach where we treat lack of pushdown for Tigris as exception.

Reply:
Ok, after your explanation I understood what was going on. I think the comment should be changed in a way that we can see what those checks are for then.

Reply:
Something like: `If tc.skipTigrisPushdown is set check that pushdown actually wasn't made.`